src/nodes/model/custom/Custom.ts

```diff
@@ -1,47 +1,65 @@
 import { Node } from '@baklavajs/core';
 import { Layers, Nodes } from '@/nodes/model/Types';
+import parse from '@/app/parser/parser';
+import OptionParams from '@/baklava/OptionParams';
 
+export enum CustomOptions{
+  InlineCode = 'Inline Code',
+}
 export default class Custom extends Node {
   type = Layers.Custom;
   name = Nodes.Custom;
 
-  private numberOfInputs = 0;
+  private inputNames: string[] = [];
+
+  private inlineCodeParams = new OptionParams(false);
 
   constructor() {
     super();
-    this.addOption('Number of Inputs', 'IntOption', 0, undefined, {
-      min: 0,
-      max: 10,
-    });
+    this.addOption(CustomOptions.InlineCode, 'TextAreaOption', '', undefined, { params: this.inlineCodeParams });
     this.addOutputInterface('Output');
     this.events.update.addListener(this, (event: any) => {
       this.nodeUpdated(event);
     });
   }
 
   private nodeUpdated(event: any) {
-    if (event.name === 'Number of Inputs') {
-      if (event.option) {
-        if (event.option.value > this.numberOfInputs) {
-          this.addInputs(event.option.value - this.numberOfInputs);
-        } else if (event.option.value < this.numberOfInputs) {
-          this.removeInputs(this.numberOfInputs - event.option.value);
-        }
+    if (event.name === CustomOptions.InlineCode) {
+      const code = event.option.value;
```

Comment:
Everytime the text changes, we update the code and most probably `parse`. What do you think about adding a debounce so we don't need to always do this? A debounce should save us parsing unnecessary code

Reply:
it's running smooth for now but could be a good idea, I'll look into it

Reply:
I agree, it could be good!

Reply:
Not to put down your good work, but whenever you type something in the node you get console errors for "badly indented file". I think a debounce would definetly be better, but also could we catch the errors from being printed by the console too?

Reply:
yeah ofc that was for debugging mainly, I'll clean it up